sklearn/manifold/tests/test_t_sne.py

```diff
@@ -764,3 +786,21 @@ def test_bh_match_exact():
     assert n_iter['exact'] == n_iter['barnes_hut']
     assert_array_almost_equal(X_embeddeds['exact'], X_embeddeds['barnes_hut'],
                               decimal=3)
+
+
+def test_tsne_with_different_distance_metrics():
+    """Make sure that TSNE works for different distance metrics"""
+    random_state = check_random_state(0)
+    n_components_original = 3
+    n_components_embedding = 2
+    X = random_state.randn(50, n_components_original).astype(np.float32)
+    metrics = ['manhattan', 'cosine']
+    dist_funcs = [manhattan_distances, cosine_distances]
+    for metric, dist_func in zip(metrics, dist_funcs):
+        tsne_1 = TSNE(metric=metric, n_components=n_components_embedding,
+                      random_state=0).fit_transform(X)
+        tsne_2 = TSNE(metric='precomputed',
+                      n_components=n_components_embedding,
+                      random_state=0).fit_transform(dist_func(X))
+        t = trustworthiness(tsne_1, tsne_2, n_neighbors=1)
```

Comment:
The trustworthiness is designed to test the validity of the results, between the input and the output space, using the euclidean distance.
You should change this test to assert that `tsne_1` and `tsne_2` are the same, .

Reply:
Done